### SQL Queries(Pune data base)

In [2]:
#number of "node" tags
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT count(*)as num from nodes;
'''
c.execute(QUERY)
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print df
db.close()

         0
0  1416239


>There are <b>1416239</b> nodes

In [54]:
# number of 'Way'tags
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT count(*)as num from ways;
'''
c.execute(QUERY)
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print df
db.close()

        0
0  270156


>There are <b>270156</b> ways

In [9]:
# number of unique users
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT DISTINCT(user) from (select user from nodes UNION ALL select user from ways);

'''
c.execute(QUERY)
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
#print df
print len(df)

db.close()

670


>There are <b>670</b> unique users 

In [14]:
#User Stats
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT user,count(user)  from (select user from nodes UNION ALL select user from ways)
group by user
order by  count(user) desc
;
'''
c.execute(QUERY) 
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print "Number of users : ",len(df)
print "\nTotal Number of Contributions : ",df[1].sum()

print "\nContribution Stats \n",df[1].describe()


db.close()


Number of users :  670

Total Number of Contributions :  1686395

Contribution Stats 
count      670.000000
mean      2517.007463
std       9387.334990
min          1.000000
25%          1.000000
50%          6.000000
75%         53.750000
max      96812.000000
Name: 1, dtype: float64


In [47]:
#Top 5 contributers
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT user,count(user)  from (select user from nodes UNION ALL select user from ways)
group by user
order by  count(user) desc
limit 5;
'''
c.execute(QUERY) 
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print df
db.close()

               0      1
0      singleton  96812
1    harishvarma  60144
2  jasvinderkaur  57697
3        sramesh  57627
4       praveeng  56788


> user "<b>singleton</b>" has contributed more than any other users

In [5]:
#amenities and their count 
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT value,count(*)as num from (select value,key from nodes_tags  UNION ALL select value,key from ways_tags)
where key='amenity'
group by value
order by num desc
;
'''
c.execute(QUERY)
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print "Number of amenities  available in PUNE CITY :", len(df)
print (df)

db.close()

Number of amenities  available in PUNE CITY : 52
                     0    1
0           restaurant  160
1               school  112
2                 fuel   69
3             hospital   56
4                 cafe   55
5              parking   49
6            fast_food   47
7                 bank   45
8              college   36
9              toilets   34
10                 atm   30
11            pharmacy   24
12              police   23
13         bus_station   22
14          university   20
15       swimming_pool   16
16              cinema   15
17    place_of_worship   15
18         post_office   14
19    community_centre    8
20      drinking_water    7
21             library    7
22            post_box    7
23             theatre    7
24                 bar    6
25        kindergarten    6
26                 pub    6
27             doctors    5
28            fountain    5
29         marketplace    5
30                taxi    5
31      coaching class    4
32        fire_station    4

In [9]:
#cusines available and their count 
import sqlite3
db = sqlite3.connect("pune.db")
c = db.cursor()
QUERY = '''SELECT value,count(*)as num from (select value,key from nodes_tags  UNION ALL select value,key from ways_tags)
where key='cuisine'
group by value
order by num desc
;
'''
c.execute(QUERY)
rows = c.fetchall()
import pandas as pd    
df = pd.DataFrame(rows)
print "Number of amenities  available in PUNE CITY :", len(df)
print (df)

db.close()

Number of amenities  available in PUNE CITY : 14
                                    0   1
0                              indian  58
1                              burger  18
2                         coffee_shop  14
3                               pizza   9
4                            sandwich   9
5                             italian   6
6                             Mastani   3
7                        North_Indian   3
8   Regional,_India,_Tandoor,_Chinese   3
9                                cafe   3
10                            chicken   3
11                            chinese   3
12                              rolls   3
13                               thai   3


>so <b>'restaurant'</b> tops the amenities list with <b>160</b>  followed by <b>"School"</b> with <b>112</b>